pages/vesting.tsx

```diff
@@ -66,6 +131,163 @@ const Vesting: NextPage = () => {
     }
   }, [activeStreams, escrowedEXA, opts, refetch]);
 
+  const { totalVestedEsEXA, totalWithdrawnEXA } = totalsStreamData;
+
+  const toogleShowAll = () => {
+    setShowAll(!showAll);
+  };
+
+  const closeCancelModal = useCallback(() => {
+    setCancelModalOpen(false);
+  }, []);
+
+  const withdrawAll = useCallback(() => {
+    if (allCancelabelStreamsIds.length === 0 && activeStreams.length > 0) {
+      track('Button Clicked', {
+        location: 'Vesting',
+        name: 'claim all',
+        value: activeStreams.length,
+      });
+      handleClaimAll();
+    }
+    if (allCancelabelStreamsIds.length > 0) {
+      track('Button Clicked', {
+        location: 'Vesting',
+        name: 'withdraw all',
+        value: allCancelabelStreamsIds.length,
+      });
+      setCancelModalOpen(true);
+    }
+  }, [activeStreams.length, allCancelabelStreamsIds.length, handleClaimAll]);
+
+  const cancelAll = useCallback(async () => {
+    if (!escrowedEXA || !opts) return;
+    setLoading(true);
+    try {
+      const tx = await escrowedEXA.write.cancel([allCancelabelStreamsIds], opts);
+      await waitForTransaction({ hash: tx });
+    } catch (e) {
+      // if request fails, don't do anything
+    } finally {
+      setLoading(false);
+      refetch();
+    }
+  }, [allCancelabelStreamsIds, escrowedEXA, opts, refetch]);
+
+  function PaperComponent(props: PaperProps | undefined) {
+    const ref = useRef<HTMLDivElement>(null);
+    return (
+      <Draggable nodeRef={ref} cancel={'[class*="MuiDialogContent-root"]'}>
+        <Paper {...props} ref={ref} />
+      </Draggable>
+    );
+  }
+
+  const Transition = React.forwardRef(function Transition(
+    props: TransitionProps & {
+      children: React.ReactElement;
+    },
+    ref: React.Ref<unknown>,
+  ) {
+    return <Slide direction="up" ref={ref} {...props} />;
+  });
+
+  const WithdrawAndCancel: React.FC<{
```

Comment:
Maybe define these 3 outside? These are being re-declared on every re-render of `Vesting`